### PR TITLE
Feat/eslint

### PR DESCRIPTION
**Pull request by manoncarbonnel**
:rotating_light: Added [eslint](https://eslint.org/) linter for `.js` `.ts` and `.tsx` files with config:

- script at root and in each component `package.json`
- html report artifact
- specified rules (according to the team's provided code guidelines)
- ignored folders containing generated files or documentation

I also fixed the errors for the **text component**.

### CODE REVIEW

**Review by dpellier — CHANGES_REQUESTED** [review]
After discussion with the team, can you change the rule to ensure space before inline bracket:
```
// OK
import { DEFAULT_ATTRIBUTE } from './constants/default-attributes';

// KO
import {DEFAULT_ATTRIBUTE} from './constants/default-attributes';
```

**Review by dpellier — CHANGES_REQUESTED** [review]
Running `yarn build:prod` locally on the examples projects fails due to eslint issues.
Do you reproduce the issue?
If so, `build:prod` is not supposed not fails due to eslint issues, can you have a look and fix it?

File: package.json [hunk 1]
```diff
@@ -95,6 +95,7 @@
     "conventional-changelog-angular": "5.0.13",
     "eslint": "8.51.0",
     "eslint-html-reporter": "0.7.4",
+    "eslint-plugin-import": "^2.29.0",
```
Comment (dpellier) [inline package.json:98]: no `^`

**Review by dpellier — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -95,7 +95,8 @@
     "conventional-changelog-angular": "5.0.13",
     "eslint": "8.51.0",
     "eslint-html-reporter": "0.7.4",
-    "eslint-plugin-import": "^2.29.0",
+    "eslint-import-resolver-typescript": "^3.6.1",
```
Comment (dpellier) [inline package.json:98]: nop

**Review by dpellier — CHANGES_REQUESTED** [review]

File: packages/components/text/src/components/osds-text/constants/default-attributes.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import type { OdsTextAttribute } from '../interfaces/attributes';
 import { ODS_TEXT_COLOR_HUE, ODS_TEXT_COLOR_INTENT } from './text-color';
 import { ODS_TEXT_LEVEL } from './text-level';
 import { ODS_TEXT_SIZE } from './text-size';
+import type { OdsTextAttribute } from '../interfaces/attributes';
```
Comment (dpellier) [inline packages/components/text/src/components/osds-text/constants/default-attributes.ts:4]: would be nice to have type on top

File: packages/components/text/src/components/osds-text/osds-text.tsx [hunk 4]
```diff
@@ -1,10 +1,11 @@
+import { odsGenerateColorVariable } from '@ovhcloud/ods-common-theming';
+import { Component, FunctionalComponent, Host, Prop, h } from '@stencil/core';
+
```
Comment (dpellier) [inline packages/components/text/src/components/osds-text/osds-text.tsx:3]: space?

**Review by dpellier — APPROVED** [review]

**Review by aesteves60 — APPROVED** [review]

**Pull request merged**